modules/storage-api/src/test/java/org/apache/ignite/internal/storage/basic/TestSortedIndexMvStorage.java

```diff
@@ -0,0 +1,249 @@
+
+/**
+ * Test implementation of MV sorted index storage.
+ */
+public class TestSortedIndexMvStorage implements SortedIndexMvStorage {
+    private final NavigableSet<BinaryRow> index;
+
+    private final SchemaDescriptor descriptor;
+
+    private final Map<Integer, TestMvPartitionStorage> pk;
+
+    private final int partitions;
+
+    private final IndexColumnView[] indexColumns;
+
+    private final int[] columnIndexes;
+
+    private final NativeType[] nativeTypes;
+
+    /**
+     * Constructor.
+     */
+    public TestSortedIndexMvStorage(
+            String name,
+            TableView tableCfg,
+            SchemaDescriptor descriptor,
+            Map<Integer, TestMvPartitionStorage> pk
+    ) {
+        this.descriptor = descriptor;
+
+        this.pk = pk;
+
+        partitions = tableCfg.partitions();
+
+        index = new ConcurrentSkipListSet<>(((Comparator<BinaryRow>) this::compareColumns).thenComparing(BinaryRow::keySlice));
+
+        // Init columns.
+        NamedListView<? extends ColumnView> tblColumns = tableCfg.columns();
+
+        TableIndexView idxCfg = tableCfg.indices().get(name);
+
+        assert idxCfg instanceof SortedIndexView;
+
+        SortedIndexView sortedIdxCfg = (SortedIndexView) idxCfg;
+
+        NamedListView<? extends IndexColumnView> columns = sortedIdxCfg.columns();
+
+        int length = columns.size();
+
+        this.indexColumns = new IndexColumnView[length];
+        this.columnIndexes = new int[length];
+        this.nativeTypes = new NativeType[length];
+
+        for (int i = 0; i < length; i++) {
+            IndexColumnView idxColumn = columns.get(i);
+
+            indexColumns[i] = idxColumn;
+
+            int columnIndex = tblColumns.namedListKeys().indexOf(idxColumn.name());
+
+            columnIndexes[i] = columnIndex;
+
+            nativeTypes[i] = SchemaDescriptorConverter.convert(SchemaConfigurationConverter.convert(tblColumns.get(columnIndex).type()));
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean supportsBackwardsScan() {
+        return true;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean supportsIndexOnlyScan() {
+        return false;
+    }
+
+    private int compareColumns(BinaryRow l, BinaryRow r) {
+        Row leftRow = new Row(descriptor, l);
+        Row rightRow = new Row(descriptor, r);
+
+        for (int i = 0; i < indexColumns.length; i++) {
+            int columnIndex = columnIndexes[i];
+
+            int cmp = PrefixComparator.compareColumns(leftRow, columnIndex, nativeTypes[i].spec(), rightRow.value(columnIndex));
+
+            if (cmp != 0) {
+                return indexColumns[i].asc() ? cmp : -cmp;
+            }
+        }
+
+        return 0;
+    }
+
+    public void append(BinaryRow row) {
+        index.add(row);
+    }
+
+    public void remove(BinaryRow row) {
+        index.remove(row);
+    }
+
+    public boolean matches(BinaryRow aborted, BinaryRow existing) {
```

Comment:
Why are the parameters called the way they are? Is it only used in the 'write abort' scenario? It just seems weird that the naming of the parameters is so highly specialized, while the name of the method is pretty generic.

Reply:
Yes, it's only used for abort right now. Should I use longer name for it?

Reply:
How about generalizing the method (renaming the parameters and make them symmetrical, that is, check both for non-nullability)?

Reply:
There's no use for this right now. It's POC, it will be heavily modified in the nearest future, that's where we'll make these changes